### ISSUE TITLE
[New Concept Exercise] : list-comprehensions

### ISSUE THREAD

**Issue by BethanyG** [exercism/python#2295]
This issue describes how to implement the `list-comprehensions` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [Contributing to Exercism](https://exercism.org/docs/building) | [Exercism and GitHub](https://exercism.org/docs/building/github) | [Contributor Pull Request Guide](https://exercism.org/docs/building/github/contributors-pull-request-guide)
- [What are those Weird Task Tags about?](https://exercism.org/docs/building/product/tasks)
- [Building Language Tracks: An Overview](https://exercism.org/docs/building/tracks) 
- [What are Concepts?](https://exercism.org/docs/building/tracks/concepts)
- [Concept Exercise Specifications](https://exercism.org/docs/building/tracks/concept-exercises)
- [Concept Specifications](https://exercism.org/docs/building/tracks/concepts) 
- [Exercism Formatting and Style Guide](https://exercism.org/docs/building/markdown/style-guide)
- [Exercism Markdown Specification](https://exercism.org/docs/building/markdown/markdown)
- [Reputation](https://exercism.org/docs/using/product/reputation)  

## Goal

The goal of this exercise is to teach the syntax and variants of `list comprehensions` in Python. 

## Learning objectives

-  Understand how a `list comprehension` relates to the `loop` + `append` method of creating/computing a `List`.
-  Create a list comprehension from a `loop` + `append`
-  Create a list comprehension from a `string`,  `List`, or other `iterable` (_such as `zip()` or `dict.items()`_)
-  Use one or more conditions/operators/methods to filter comprehension inputs
-  Use methods or logic to format the elements (output members) of the comprehension
-  Create a _nested comprehension_ 
-  Create a _nested comprehension_ with one or more formatting or filtering conditions

## Out of scope

- Memory and performance characteristics and optimizations
- `generators` and `generator expressions` in `list comprehensions`
-  How `map()` and `filter()` relate to or differ from `list comprehensions`
-  Using the `assignment expression` (_walrus operator_) with a `list comprehension`

## Concepts

-  `list-comprehensions`

## Prerequisites

- `basics`
-  `bools`
- `comparisons`  
- `conditionals`
- `lists`
- `list-methods`
- `loops`
- `iteration`
- `sequences`

## Resources to refer to

-  [List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- [Nested List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#nested-list-comprehensions)
- [When to Use a List Comprehension in Python (Real Python)](https://realpython.com/list-comprehension-python/)
- [Python List Comprehensions:  Explained Visually (Trey Hunner)](https://treyhunner.com/2015/12/python-list-comprehensions-now-in-color/)

### Hints

- `List Comprehensions` section of the Python docs tutorial:  [List Comprehensions](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- This animated GIF from Trey Hunner:  [List Comprehensions: The Movie](https://treyhunner.com/images/list-comprehension-condition.gif)

### After

-  `comprehension syntax` for other data structures such as `sets` and `dictionaries`
- `generators` and `generator expressions`
- `generators` and `generator expressions` in `list comprehensions`

## Representer

No changes required.

## Analyzer

No changes rquired.

## Implementing

- Tests should be written using unittest.TestCase, and the test file named list_comprehensions_test.py.
- [How to Implement a Concept Exercise in Python](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md)
- [make-concept-exercise Utility](https://github.com/exercism/v3/tree/master/languages/python/bin)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by Steffan153** [comment]
Is this actually claimed? :) If not, I'd like to work on it. By the way, can we steal https://github.com/exercism/elixir/blob/main/exercises/concept/boutique-suggestions/.docs/instructions.md for this? Elixir's list comprehensions are very similar.

**Comment by BethanyG** [comment]
Hi @Steffan153!  Welcome to the Python track.  👋🏽   
This is indeed already claimed by me, and is in progress.  Thanks for checking tho!

**Comment by BethanyG** [comment]
Flagging this as a re-do with new issue template, as this has aged, and the actions bot is getting annoying.  Closing for now until the redo can happen.
